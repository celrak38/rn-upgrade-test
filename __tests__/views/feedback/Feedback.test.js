import 'react-native'
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

// Component to be tested
import { FeedbackView } from '../../../app/views/preferences/feedback/Feedback'

Enzyme.configure({ adapter: new Adapter() })

// Mock props passed down from state
const initialState = {
	timeoutFeedback: jest.fn(),
	feedback: {
		comment: '',
		name: '',
		email: '',
	}
}

// Set up component to be rendered
function setup(props) {
	return shallow(<FeedbackView {...props} />)
}

test('renders without crashing', () => {
	const tree = setup(initialState)
	expect(tree).toMatchSnapshot()
})

test('loading screen renders when requesting', () => {
	const loadingState = initialState
	loadingState.requestStatus = { timeRequested: new Date() }
	const tree = setup(loadingState)
	expect(tree).toMatchSnapshot()
})

test('renders form when request is finished', () => {
	// Loading state while feedback posts
	const loadingState = initialState
	loadingState.requestStatus = { timeRequested: new Date() }

	const tree = shallow(<FeedbackView {...loadingState} />)

	// Successful post state
	const successState = initialState
	successState.requestStatus = null
	successState.feedback.response = {}

	tree.setProps(successState)
	expect(tree).toMatchSnapshot()

	// Failed post state
	const failingState = initialState
	failingState.requestError = {}

	tree.setProps(failingState)
	expect(tree).toMatchSnapshot()
})
