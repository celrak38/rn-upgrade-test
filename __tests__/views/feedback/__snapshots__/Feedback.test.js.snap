// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading screen renders when requesting 1`] = `
<ScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 16,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "paddingBottom": 16,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Your feedback is being submitted...
    </Text>
  </View>
</ScrollView>
`;

exports[`renders form when request is finished 1`] = `
<_class
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardOpeningTime={250}
  viewIsInsideTabBar={false}
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "marginBottom": 8,
          "marginHorizontal": 8,
          "marginTop": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flexWrap": "wrap",
            "fontSize": 18,
            "lineHeight": 24,
            "paddingBottom": 16,
          }
        }
      >
        Help us make the 
        Campus Mobile
         app better.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flexWrap": "wrap",
            "fontSize": 18,
            "lineHeight": 24,
            "paddingBottom": 16,
          }
        }
      >
        Submit your thoughts and suggestions here.
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#EAEAEA",
            "flexDirection": "row",
            "marginBottom": 8,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          blurOnSubmit={true}
          maxLength={500}
          multiline={true}
          onChange={[Function]}
          placeholder="Tell us what you think*"
          returnKeyType="done"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flex": 1,
              "fontSize": 18,
              "padding": 8,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#EAEAEA",
            "marginBottom": 8,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          maxLength={100}
          onChange={[Function]}
          placeholder="Email"
          returnKeyType="done"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flex": 1,
              "fontSize": 18,
              "padding": 8,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        underlayColor="rgba(200,200,200,.1)"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#034262",
              "borderRadius": 3,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 16,
              }
            }
          >
            Submit
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </TouchableWithoutFeedback>
</_class>
`;

exports[`renders form when request is finished 2`] = `
<_class
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardOpeningTime={250}
  viewIsInsideTabBar={false}
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "marginBottom": 8,
          "marginHorizontal": 8,
          "marginTop": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flexWrap": "wrap",
            "fontSize": 18,
            "lineHeight": 24,
            "paddingBottom": 16,
          }
        }
      >
        Help us make the 
        Campus Mobile
         app better.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flexWrap": "wrap",
            "fontSize": 18,
            "lineHeight": 24,
            "paddingBottom": 16,
          }
        }
      >
        Submit your thoughts and suggestions here.
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#EAEAEA",
            "flexDirection": "row",
            "marginBottom": 8,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          blurOnSubmit={true}
          maxLength={500}
          multiline={true}
          onChange={[Function]}
          placeholder="Tell us what you think*"
          returnKeyType="done"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flex": 1,
              "fontSize": 18,
              "padding": 8,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#EAEAEA",
            "marginBottom": 8,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          maxLength={100}
          onChange={[Function]}
          placeholder="Email"
          returnKeyType="done"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flex": 1,
              "fontSize": 18,
              "padding": 8,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        underlayColor="rgba(200,200,200,.1)"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#034262",
              "borderRadius": 3,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 16,
              }
            }
          >
            Submit
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </TouchableWithoutFeedback>
</_class>
`;

exports[`renders without crashing 1`] = `
<_class
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardOpeningTime={250}
  viewIsInsideTabBar={false}
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "marginBottom": 8,
          "marginHorizontal": 8,
          "marginTop": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flexWrap": "wrap",
            "fontSize": 18,
            "lineHeight": 24,
            "paddingBottom": 16,
          }
        }
      >
        Help us make the 
        Campus Mobile
         app better.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flexWrap": "wrap",
            "fontSize": 18,
            "lineHeight": 24,
            "paddingBottom": 16,
          }
        }
      >
        Submit your thoughts and suggestions here.
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#EAEAEA",
            "flexDirection": "row",
            "marginBottom": 8,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          blurOnSubmit={true}
          maxLength={500}
          multiline={true}
          onChange={[Function]}
          placeholder="Tell us what you think*"
          returnKeyType="done"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flex": 1,
              "fontSize": 18,
              "padding": 8,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#EAEAEA",
            "marginBottom": 8,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          maxLength={100}
          onChange={[Function]}
          placeholder="Email"
          returnKeyType="done"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "flex": 1,
              "fontSize": 18,
              "padding": 8,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        underlayColor="rgba(200,200,200,.1)"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#034262",
              "borderRadius": 3,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 16,
              }
            }
          >
            Submit
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </TouchableWithoutFeedback>
</_class>
`;
